src/pybind/mgr/cephadm/templates/services/ingress/haproxy.cfg.j2

```diff
@@ -48,7 +48,7 @@ defaults
 frontend stats
     mode http
     bind {{ ip }}:{{ monitor_port }}
-    bind localhost:{{ monitor_port }}
+    bind {{ local_host_ip }}:{{ monitor_port }}
```

Comment:
I don't know this 100% but I think this is actually causing some teuthology test failures. From what I can gather, the beginning of the keepalived conf
```
vrrp_script check_backend {
    script "/usr/bin/curl http://localhost:9002/health"
    weight -20
    interval 2
    rise 2
    fall 2
}
```
is trying to check for `localhost:9002` but since haproxy is now bound to the this other local host ip the keepalived isn't properly checking on the haproxy. This means if the haproxy goes down keepalived never reports it and the other haproxy never becomes active. This was causing tests to timeout waiting on a mount point to be accessible because it was expecting the other haproxy to take over when one was stopped but it never happened. Changing the keepalived conf to use the correct IP and restarting the service allowed the tests to run to completion. It seems you may need to update the keepalive conf with this IP as well.

Examples of jobs that timed out waiting for mount point to be accessible:
https://pulpito.ceph.com/adking-2023-01-12_12:11:23-orch:cephadm-wip-adk-testing-2023-01-11-2322-distro-default-smithi/7135288
https://pulpito.ceph.com/adking-2023-01-12_12:11:23-orch:cephadm-wip-adk-testing-2023-01-11-2322-distro-default-smithi/7135295

Reply:
@adk3798 your observations are right. Keepalived was listening on the wrong ip (`localhost`) I changed the code to listen on the correct ip (and adjusted the UT accordingly).